overlord/devicestate/handlers_install.go

```diff
@@ -361,6 +361,19 @@ func (m *DeviceManager) doSetupRunSystem(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// TODO: FIXME: this should go away after we have time to design a proper
+	//              solution
+	// TODO: only run on specific models?
+
+	// on some specific devices, we need to create these directories in
+	// _writable_defaults in order to allow the install-device hook to install
+	// some files there, this eventually will go away when we introduce a proper
+	// mechanism not using system-files to install files onto the root
+	// filesystem from the install-device hook
+	if err := fixupWritableDefaultDirs(boot.InstallHostWritableDir); err != nil {
+		return err
```

Comment:
the order of this doesn't really matter I think, it just needs to be run before the install-device hook is executed

Reply:
Why is it safer though? if we get rebooted before we execute `bootEnsureNextBootToRunMode` in `doRestartSystemToRunMode` then we will just redo everything we have done here up to this point

Reply:
because making the system bootable writes the modenv and seals, as the input to those two things evolves over the time it seems best to not touch much of the system after doing that

Reply:
okay fair enough